Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,25 +1,113 @@
+import { useState } from "react";
+
 import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
+  const [visible, setVisible] = useState(false);
+  type InputState = {
+    speed: number | undefined;
+    battery: number | undefined;
+    weather: number;
+  };
+  const [inputs, setInputs] = useState<InputState>({
+    speed: 0,
+    weather: 50,
+    battery: 100,
+  });
+
   const calculateRange = () => {
-    return;
-  }
+    event?.preventDefault();
+    setVisible(true);
+  };
+
+  const BatteryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = parseInt(event.target.value);
+    setVisible(false);
+    if (value) {
+      setInputs({ ...inputs, battery: value });
+    } else {
+      setInputs({ ...inputs, battery: undefined });
+    }
+  };
+
+  const SpeedChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = parseInt(event.target.value);
+    setVisible(false);
+    if (value !== undefined) {
+      setInputs({ ...inputs, speed: value });
+    } else {
+      setInputs({ ...inputs, speed: undefined });
+    }
+  };
+
+  const sliderChange = (num: string) => {
+    const value = parseFloat(num);
+    setInputs({ ...inputs, weather: value });
+    setVisible(false);
+  };
```

Comment:
in all 3 of these functions, you are doing the same thing,     setVisible(false);

can you leverage useReducer and create 3 actions which will update each of the respective inputs?